tsdb/index/tsi1/log_file.go

```diff
@@ -1177,30 +1193,96 @@ type logMeasurement struct {
 	tagSet  map[string]logTagKey
 	deleted bool
 	series  map[uint64]struct{}
+	bitmap  *tsdb.SeriesIDSet
 }
 
 // bytes estimates the memory footprint of this logMeasurement, in bytes.
-func (mm *logMeasurement) bytes() int {
+func (m *logMeasurement) bytes() int {
 	var b int
-	b += len(mm.name)
-	for k, v := range mm.tagSet {
+	b += len(m.name)
+	for k, v := range m.tagSet {
 		b += len(k)
 		b += v.bytes()
 	}
-	b += len(mm.series) * 8
-	b += int(unsafe.Sizeof(*mm))
+	b += (int(m.cardinality()) * 8)
+	b += int(unsafe.Sizeof(*m))
 	return b
 }
 
-func (mm *logMeasurement) seriesIDs() []uint64 {
-	a := make([]uint64, 0, len(mm.series))
-	for seriesID := range mm.series {
+func (m *logMeasurement) addSeriesID(x uint64) {
+	if m.bitmap != nil {
+		m.bitmap.AddNoLock(x)
+		return
+	}
+
+	m.series[x] = struct{}{}
+
+	// If the map is getting too big it can be converted into a roaring bitmap.
+	if len(m.series) > 25 {
+		m.bitmap = tsdb.NewSeriesIDSet()
+		for id := range m.series {
+			m.bitmap.AddNoLock(id)
+		}
+		m.series = nil
+	}
```

Comment:
I'd be curious to see what the impact is in practice for `<25` series sets. It seems like it's going to make the code a lot more complicated by handling both cases.

Reply:
I have some benchmarks somewhere. I plan on abstracting all this away into the `SeriesIDSet` implementation, so we can make changes once and they'll apply everywhere.